### PR TITLE
Add e2e test and refactor

### PR DESCRIPTION
**Pull request by francois-potato**
* Add e2e test
* Refactor publishing code
* Modify exports

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
   "workspaces": [
     "packages/api",
     "packages/client",
+    "packages/e2e",
```
Comment (joshJarr) [inline package.json:9]: Looks like this package is omitted in this commit, is this intentional @francois-potato?

Comment (francois-potato) [reply package.json:9]: Typical; I forgot to push the new files!

**Review by joshJarr — APPROVED** [review]

File: packages/e2e/package.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "@w3name/e2e",
+  "version": "0.1.0",
+  "description": "e2e tests for w3name",
+  "author": "",
+  "private": true,
+  "license": "Apache-2.0 OR MIT",
+  "scripts": {
+    "test": "npx mocha --experimental-modules --experimental-vm-modules test/**/*.spec.mjs"
```
Comment (joshJarr) [inline packages/e2e/package.json:9]: Is it worth running these tests in CI too?

Comment (francois-potato) [reply packages/e2e/package.json:9]: 👍

**Pull request merged**